HARJA-1451 Luo mhu-muutosten alustava tietomalli, näkymärunko ja testidataa

jarnovayrynen opened the pull request.

jarnovayrynen committed:
HARJA-1451 Luo mhu-muutosten alustava tietomalli ja testidataa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
Lisää palvelu josta haetaan muutoksia

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
HARJA-1367	Muutokset-näkymän runko

jarnovayrynen committed:
Laske muutoksen vaikutus tavoitehintaan

jarnovayrynen committed:
Luo collapsoitavan taulukon komponentti

K1ll3rF0x commented on tietokanta/src/main/resources/db/migration/V1_2170__.sql:
```diff
@@ -0,0 +1,138 @@
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER mhu_muutos_historia_trigger
```
Tämän pitäisi olla varmaan nimeltään:
```suggestion
CREATE TRIGGER mhu_muutos_liite_historia_trigger
```

mhu_muutos_historia_trigger on määritelty tuolla ylempänä.

jarnovayrynen committed:
HARJA-1367 Viimeistele Muutokset-näkymän runko

jarnovayrynen committed:
HARJA-1367 Viimeistele Muutokset-näkymän runko

Lisää runkoon vaihtoehtoiset komponentit, jos ollaan hoitokaudessa 2024-2025 tai aiemmin. Silloin näytetään tavoitehinnan muutosket ja suunniteltujen määrien muutokset eri tavalla.

jarnovayrynen committed:
Korjaa docstring tiedonhaun funktiossa

jarnovayrynen committed:
Lisää mhu_muutos tauluun validi aika range

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
Edistä pysyvää muutosta

jarnovayrynen committed:
Korjaa triggerin nimi

jarnovayrynen committed:
Lisää muutoksen lomakerunkoa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-1451_muutosten_tietomalli

jarnovayrynen committed:
Vaihda migraationumero

jarnovayrynen committed:
Siisti muutoksen tallennuksen stubi

jarnovayrynen committed:
Siisti lokeja

jarnovayrynen committed:
Lisää muutokset-komponentti

jarnovayrynen committed:
Lisää muutokset-komponentti

jarnovayrynen committed:
Lisää muutosten vaiktuksien yhteenveto -komponentti

samnes commented on tietokanta/src/main/resources/db/migration/V1_1172__.sql:
```diff
@@ -0,0 +1,142 @@
+-- voi olla mhu_muutoksia monta riviä samalla id:lla kuten halutaankin
+CREATE TABLE mhu_muutos_historia () INHERITS (mhu_muutos);
+-- varmistetaan että samaa (id, versio)-yhdistelmää on vain yksi
+CREATE UNIQUE INDEX mhu_muutos_historia_id_versio ON mhu_muutos_historia (id, versio);
```
Tarvitaanko muille historiatauluille varmistusta uniikkiudesta?
